### PR TITLE
Update Canary to v1.5.2

### PR DESCRIPTION
**Pull request by schjonhaug**
## Summary
Update Canary myNode app package to v1.5.2.

This continues the previous v1.5.1 review from #1004 and includes the fixes made in response to that feedback.

## Release summary
Canary v1.5.2 rejects invalid xpub descriptors before they can create wallets that cannot sync correctly, and adds recovery handling so failed or stuck wallet records can be deleted and recreated. It also fixes Genesis wallet/address-watch syncing edge cases, and adds myNode local transaction explorer detection when Mempool or BTC RPC Explorer is available.

## Upstream release
https://github.com/schjonhaug/canary/releases/tag/v1.5.2

## Test plan
- [x] Built with `mynode-sdk build canary`
- [x] Tested on local myNode

### CODE REVIEW

File: rootfs/standard/usr/share/mynode_apps/canary/scripts/install_canary.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ set -e
 
 echo "==================== INSTALLING APP ===================="
 
-VERSION="${VERSION:-v1.5.0}"
+VERSION="${VERSION:-v1.5.2}"
```
Comment (mynode-dev) [inline rootfs/standard/usr/share/mynode_apps/canary/scripts/install_canary.sh]: Are you still having issues if you don't re-define this? It should be present in the environment already.

File: rootfs/standard/usr/share/mynode_apps/canary/app_data/docker-compose.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ version: "3.8"
 
 services:
   backend:
-    image: canary-backend:latest
+    image: schjonhaug/canary-backend:v1.5.2
```
Comment (mynode-dev) [inline rootfs/standard/usr/share/mynode_apps/canary/app_data/docker-compose.yml]: Could you make this a variable that gets set as part of an environment. Most apps can get down to have the only version specifier necessary be the one in the apps JSON data.

For example, you could do what mempool-docker-compose.yml does and use ${VERSION} with a .env file generated in the pre_canary.sh script (like pre_mempool.sh).

Alternatively, you could delete and re-tag the latest version of canary as latest (which you seem to be doing already) and keep the latest reference in this file.

File: rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh [hunk 3]
```diff
@@ -12,6 +12,21 @@ generate_secret() {
     tr -dc A-Za-z0-9 < /dev/urandom | head -c "$length"
 }
 
+service_is_enabled() {
```
Comment (mynode-dev) [inline rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh]: These could be helpful. Do you want to move these to mynode_functions.sh?

At least move service_is_enabled() and then rename it to is_service_enabled() to match the python version. The other two I may want to remove. I'll explain in another comment.

File: rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh [hunk 4]
```diff
@@ -30,5 +45,21 @@ CANARY_SELF_HOSTED_ADMIN_PASSWORD=$(cat "$ADMIN_PASSWORD_FILE")
 JWT_SECRET=$(cat "$JWT_SECRET_FILE")
 EOF
 
+has_local_tx_explorer=0
+
+if service_is_available mempool; then
```
Comment (mynode-dev) [inline rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh]: Using service_is_available() may cause a race condition on startup. A number of apps start at the same time so Canary may not realize the other apps are available because they are not yet active. I think sticking with an is_service_enabled() check only would be a better approach.

Was there a reason you had to add the check for "is-active"?

File: rootfs/standard/usr/share/mynode_apps/canary/scripts/install_canary.sh (hunk 1 above)
Comment (schjonhaug) [reply rootfs/standard/usr/share/mynode_apps/canary/scripts/install_canary.sh]: You're right — removed. `VERSION` is already injected into the install script's environment by the framework (from `canary.json`'s `latest_version`), so the `${VERSION:-…}` fallback was redundant. The script now just uses the framework-provided `VERSION`.

File: rootfs/standard/usr/share/mynode_apps/canary/app_data/docker-compose.yml (hunk 2 above)
Comment (schjonhaug) [reply rootfs/standard/usr/share/mynode_apps/canary/app_data/docker-compose.yml]: Went with your second suggestion and reverted these back to `canary-backend:latest` / `canary-frontend:latest`. `install_canary.sh` already pulls the pinned `:$VERSION` image and retags it as `:latest`, so the only version specifier left is the one in `canary.json`. (I didn't take the `.env`/`${VERSION}` route since the pre/runtime scripts don't receive `VERSION` from the framework the way install scripts do, so it'd need extra plumbing for a dynamic app — the retag path keeps it simpler.)

File: rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh (hunk 3 above)
Comment (schjonhaug) [reply rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh]: Moved `is_service_enabled()` into `mynode_functions.sh` (renamed to match the python `is_service_enabled`) and `pre_canary.sh` now sources it. I also dropped the other two helpers (`service_is_active`/`service_is_available`) entirely per your next comment.

File: rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh (hunk 4 above)
Comment (schjonhaug) [reply rootfs/standard/usr/share/mynode_apps/canary/scripts/pre_canary.sh]: Good catch, and thanks for the explanation. The `is-active` check was the tightening I added in #1004 to handle a service being stopped after Canary starts — but you're right that at startup it races against other apps still coming up. Reverted to an `is_service_enabled`-only check, which is why the `service_is_active`/`service_is_available` helpers are gone.

**Review by mynode-dev — APPROVED** [review]

**Comment by mynode-dev** [comment]
Looks good. Thanks!
